### PR TITLE
Add error messages when wrong pin is entered

### PR DESCRIPTION
**Pull request by Tbaut**
- fix https://github.com/paritytech/parity-signer/issues/156
- added error messages at pin creation when pin is too short or don't match
- added an error message at account unlock

Screenshots:
image
image
image

### CODE REVIEW

File: src/screens/AccountUnlock.js [hunk 1]
```diff
@@ -118,31 +127,49 @@ export class AccountUnlockAndChangePin extends React.PureComponent {
...
+    console.log('verifyAndSubmitPin, pin: ' + this.state.pin + ', submitted: ' + submitted)
+    debounce(this.props.onChange, 200)({
+      pin: this.state.pin,
+      submitted: submitted
+    });
+  }
+
   render() {
-    const { accounts } = this.props;
+    console.log("hasWrongPin: " + this.props.hasWrongPin)
```
Comment (fckt) [inline src/screens/AccountUnlock.js]: unnecessary `console.log`

File: src/screens/AccountUnlock.js [hunk 2]
```diff
@@ -118,31 +127,49 @@ export class AccountUnlockAndChangePin extends React.PureComponent {
...
         <Background />
         <Text style={styles.titleTop}>UNLOCK ACCOUNT</Text>
+        <Text style={styles.errorText}>{this.props.hasWrongPin == undefined || !this.props.hasWrongPin ? null : 'Wrong pin, please try again'}</Text>
         <Text style={styles.title}>PIN</Text>
         <PinInput
-          onChangeText={pin => {
+          onChangeText={(pin) => {
+            const submitted = false
             this.setState({ pin });
+            console.log("TextChange: " + this.state.pin + " Submitted: " + submitted)
```
Comment (fckt) [inline src/screens/AccountUnlock.js]: unnecessary `console.log`

File: src/screens/AccountUnlock.js [hunk 3]
```diff
@@ -118,31 +127,49 @@ export class AccountUnlockAndChangePin extends React.PureComponent {
...
             if (pin.length < 1) {
               return;
             }
-            debounce(this.props.onChange, 200)(pin);
+            this.verifyAndSubmitPin(submitted);
+          }}
+          onSubmitEditing={() => {
+            const submitted = true
+            console.log("Text: " + this.state.pin + " submittedChanged : " + submitted)
```
Comment (fckt) [inline src/screens/AccountUnlock.js]: unnecessary `console.log`

**Comment by fckt** [comment]
cool! Could you attach some screenshots also?

File: src/screens/AccountUnlock.js [hunk 4]
```diff
@@ -85,8 +87,13 @@ export class AccountUnlock extends React.Component {
                   ]
                 });
                 this.props.navigation.dispatch(resetAction);
+                console.log('verifyPinAndNavigate 1')
```
Comment (fckt) [inline src/screens/AccountUnlock.js]: unnecessary `console.log`

File: src/screens/AccountUnlock.js [hunk 5]
```diff
@@ -85,8 +87,13 @@ export class AccountUnlock extends React.Component {
...
+              } else if (o.submitted != undefined && o.submitted) {
+                this.setState({ hasWrongPin: true });
+                console.log('Submitted a wrong pin')
```
Comment (fckt) [inline src/screens/AccountUnlock.js]: unnecessary `console.log`

File: src/screens/AccountUnlock.js [hunk 6]
```diff
@@ -118,31 +127,49 @@ export class AccountUnlockAndChangePin extends React.PureComponent {
...
             this.setState({ pin });
+            console.log("TextChange: " + this.state.pin + " Submitted: " + submitted)
             if (pin.length < 1) {
               return;
             }
-            debounce(this.props.onChange, 200)(pin);
+            this.verifyAndSubmitPin(submitted);
+          }}
+          onSubmitEditing={() => {
+            const submitted = true
```
Comment (fckt) [inline src/screens/AccountUnlock.js]: unnecessary constant

File: src/screens/AccountUnlock.js [hunk 7]
```diff
@@ -118,31 +127,49 @@ export class AccountUnlockAndChangePin extends React.PureComponent {
...
 
   static propTypes = {
     onChange: PropTypes.func.isRequired,
-    nextButtonTitle: PropTypes.string
+    nextButtonTitle: PropTypes.string,
+    hasWrongPin: PropTypes.bool
   };
 
+  verifyAndSubmitPin = (submitted) => {
+    console.log('verifyAndSubmitPin, pin: ' + this.state.pin + ', submitted: ' + submitted)
```
Comment (fckt) [inline src/screens/AccountUnlock.js]: unnecessary `console.log`

File: src/stores/AccountsStore.js [hunk 8]
```diff
@@ -18,10 +18,9 @@
 
 import { Container } from 'unstated';
 import debounce from 'debounce';
-import { loadAccounts, saveAccount, deleteAccount } from '../util/db';
+import { loadAccounts, saveAccount } from '../util/db';
```
Comment (fckt) [inline src/stores/AccountsStore.js]: The rule is: one feature (logical change) - one PR. Make a separate PR for these changes, please.

File: src/stores/AccountsStore.js [hunk 9]
```diff
@@ -117,7 +116,7 @@ export default class AccountsStore extends Container<AccountsState> {
     });
   }
 
-  async loadAccountTxs() {}
+  async loadAccountTxs() { }
```
Comment (fckt) [inline src/stores/AccountsStore.js]: The rule is: one feature (logical change) - one PR. Make a separate PR for these changes, please.

File: src/stores/ScannerStore.js [hunk 10]
```diff
@@ -17,10 +17,9 @@
 // @flow
 import { Container } from 'unstated';
 import transaction from '../util/transaction';
-import { keccak, ethSign, brainWalletSign, decryptData } from '../util/native';
-import { saveTx, loadAccountTxs } from '../util/db';
-import { accountId } from '../util/account';``
-import { NETWORK_TITLES, NETWORK_IDS } from '../constants';
+import { keccak, brainWalletSign, decryptData } from '../util/native';
```
Comment (fckt) [inline src/stores/ScannerStore.js]: The rule is: one feature (logical change) - one PR. Make a separate PR for these changes, please.

**Comment by Tbaut** [comment]
>Did some cleanup for unused components or functions (sometime I just commented it, will remove it if it's ok, along with the console.log)
^ the description of the issue :) I'll remove those `console.log` of course, it's a WIP requesting feedback in general.

File: src/stores/AccountsStore.js (hunk 9 above)
Comment (Tbaut) [reply src/stores/AccountsStore.js]: That's a linter cleanup, I'm ok with other rules, I guess we can accept that right?

File: src/stores/ScannerStore.js (hunk 10 above)
Comment (Tbaut) [reply src/stores/ScannerStore.js]: :+1:

**Comment by Tbaut** [comment]
It should be good to go if you want to have a look :)

**Comment by fckt** [comment]
As we discussed before, we's better to not change the current `onChange` behavior. What we decided to do is to just show user the current "unlocking" status.

**Comment by Tbaut** [comment]
Here we go, I've moved the error message to the state.
As we discussed, I don't think it's possible to leave the `onChange` function unchanged as I need to pass the information if the pin is wrong or right. If should now work well on iphone as well.

Let me know your thoughts.

File: src/screens/AccountUnlock.js [hunk 11]
```diff
@@ -93,29 +100,42 @@ export class AccountUnlock extends React.Component {
...
         <Background />
         <Text style={styles.titleTop}>UNLOCK ACCOUNT</Text>
+        <Text style={styles.errorText}>{this.showErrorMessage()}</Text>
         <Text style={styles.title}>PIN</Text>
         <PinInput
-          onChangeText={pin => {
-            this.setState({ pin });
-            if (pin.length < 1) {
+          onChangeText={async (pin) => {
+            this.setState({ pin: pin })
```
Comment (fckt) [inline src/screens/AccountUnlock.js]: could be leaved as just `this.setState({ pin });`
